src/util/simplify_expr_class.h

```diff
@@ -62,6 +64,15 @@ class simplify_exprt
 
   bool do_simplify_if;
 
+  enum class was_changedtt
+  {
+    Changed, Unchanged
+  };
+  struct simplify_resultt
+  {
+    exprt new_expr;
+    was_changedtt was_changed;
+  };
```

Comment:
I had also considered using `optionalt` instead of introducing these two types. On the one hand reusing generic data structures is nice, but on the other hand I feel it might be easier to see what's going on if we spell it out with the `Changed`/`Unchanged` thing.

Reply:
I'm certainly happy with either approach - `optionalt` captures the 'you either get something that has changed back, or nothing' semantics, but a new type *may* be clearer. No strong opinions from me here.

Reply:
I am currently looking at this. The benefit of the pair expression/was_changed is that functions returning that can be chained, i.e., you can write
`simplify_X(simplify(Y(whatnot))`
With `std::optional` that needs a branch.